react/src/hooks/analyses/useAnalysesPage.tsx

```diff
@@ -14,9 +15,40 @@ async function fetchAnalyses(query: Query<Analysis>) {
 export function useAnalysesPage() {
     const queryClient = useQueryClient();
 
-    return async (query: Query<Analysis>) => {
-        return await queryClient.fetchQuery<QueryResult<Analysis>, Error>(["analyses", query], () =>
-            fetchAnalyses(query)
-        );
-    };
+    const func = useCallback(
+        async (query: Query<Analysis>) => {
+            // If analysis_id is specified, we'll use the individual GET endpoint
+            const idFilter = query.filters.find(
+                filter => filter.column.field === "analysis_id" && filter.value
+            );
+            if (idFilter && typeof idFilter.value === "string") {
```

Comment:
Is the type not well-defined?

Reply:
material-table types the filter values as "any". The string type-check is just to avoid unexpected behaviour and to make the filter value types play nice in the if-block.